File: backend/app/schemas/setup_progress.py
"""Dashboard facility setup checklist — derived counts per tenant."""

from __future__ import annotations

from pydantic import BaseModel, Field


class SetupProgressOut(BaseModel):
    blueprint_count: int = Field(ge=0)
    zone_count: int = Field(ge=0)
    equipment_count: int = Field(ge=0)
    worker_user_count: int = Field(ge=0, description="Active users with role=worker in this company.")
    procedure_task_count: int = Field(ge=0, description="Project tasks (procedures) for this company.")
    gateway_count: int = Field(ge=0, description="Registered automation gateways.")
    ble_device_count: int = Field(ge=0, description="Registered BLE tags / devices.")
    work_request_count: int = Field(ge=0, description="All work requests for this company.")

    facility_layout_done: bool
    zones_done: bool
    equipment_done: bool
    workers_done: bool
    procedures_done: bool
    devices_done: bool
    maintenance_started_done: bool

File: frontend/lib/setup-progress.ts
import { apiFetch } from "@/lib/api";

export type SetupProgressState = {
  blueprint_count: number;
  zone_count: number;
  equipment_count: number;
  worker_user_count: number;
  procedure_task_count: number;
  gateway_count: number;
  ble_device_count: number;
  work_request_count: number;
  facility_layout_done: boolean;
  zones_done: boolean;
  equipment_done: boolean;
  workers_done: boolean;
  procedures_done: boolean;
  devices_done: boolean;
  maintenance_started_done: boolean;
};

export async function fetchSetupProgress(): Promise<SetupProgressState> {
  return apiFetch<SetupProgressState>("/api/v1/setup-progress");
}
